src/otx/v2/adapters/torch/mmengine/mmaction/engine.py

```diff
@@ -0,0 +1,302 @@
+"""OTX adapters.torch.mmengine.mmaction.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import copy
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import torch
+from mmaction.registry import VISUALIZERS
+
+from otx.v2.adapters.torch.mmengine.engine import MMXEngine
+from otx.v2.adapters.torch.mmengine.mmaction.registry import MMActionRegistry
+from otx.v2.adapters.torch.mmengine.mmdeploy import AVAILABLE
+from otx.v2.adapters.torch.mmengine.modules.utils.config_utils import CustomConfig as Config
+from otx.v2.api.utils.logger import get_logger
+
+if TYPE_CHECKING:
+    import numpy as np
+
+
+logger = get_logger()
+
+MMACTION_DEFAULT_CONFIG_PATH = "src/otx/v2/configs/action_classification/otx_mmaction_classification_default.yaml"
+MMACTION_DEFAULT_CONFIG = Config.fromfile(MMACTION_DEFAULT_CONFIG_PATH)
+
+class MMActionEngine(MMXEngine):
+    """The MMActionEngine class is responsible for running inference on pre-trained models."""
+
+    def __init__(
+        self,
+        work_dir: str | Path | None = None,
+    ) -> None:
+        """Initialize a new instance of the MMActionEngine class.
+
+        Args:
+            work_dir (Optional[Union[str, Path]], optional): The working directory for the engine. Defaults to None.
+            config (Optional[Union[Dict, Config, str]], optional): The configuration for the engine. Defaults to None.
+        """
+        super().__init__(work_dir=work_dir)
+        self.registry = MMActionRegistry()
+
+    def _update_eval_config(self, evaluator_config: list | dict | None) -> list | dict | None:
+        if evaluator_config is None or not evaluator_config:
+            evaluator_config = {
+                "type": "AccMetric",
+                "metric_list": ("top_k_accuracy", "mean_class_accuracy"),
+            }
+        return evaluator_config
+
+    def _update_config(
+        self,
+        func_args: dict,
+        **kwargs,
+    ) -> tuple[Config, bool]:
+        config, update_check = super()._update_config(func_args, **kwargs)
+
+        for subset in ("val", "test"):
+            if f"{subset}_dataloader" in config and config[f"{subset}_dataloader"] is not None:
+                evaluator_config = self._get_value_from_config(f"{subset}_evaluator", func_args)
+                config[f"{subset}_evaluator"] = self._update_eval_config(
+                    evaluator_config=evaluator_config,
+                )
```

Comment:
This one could fit too?
```suggestion
                config[f"{subset}_evaluator"] = self._update_eval_config(evaluator_config=evaluator_config)
```

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/7169459c0e1c71d2ee480b906c4533af36004ca9